Network/HTTP/Semantics/FillBuf.hs

```diff
@@ -122,14 +130,14 @@ type NextWithTotal = Int -> DynaNext
 runStreamingChunk :: StreamingChunk -> NextWithTotal -> NextWithTotal
 runStreamingChunk chunk next =
     case chunk of
-        StreamingFinished dec -> finished dec
+        StreamingFinished mdec -> finished mdec
         StreamingFlush -> flush
-        StreamingBuilder builder Nothing -> runStreamingBuilder builder next
-        StreamingBuilder builder (Just dec) -> runStreamingBuilder builder (finished dec)
+        StreamingBuilder builder NotEndOfStream -> runStreamingBuilder builder next
+        StreamingBuilder builder (EndOfStream mdec) -> runStreamingBuilder builder (finished mdec)
   where
-    finished :: CleanupStream -> NextWithTotal
-    finished dec = \total _buf _room -> do
-        dec
+    finished :: Maybe CleanupStream -> NextWithTotal
+    finished mdec = \total _buf _room -> do
+        fromMaybe (return ()) mdec
         return $ Next total True Nothing
```

Comment:
In the future, it might make sense to avoid requesting a flush here if `total == 0`, to prevent *any* empty data frames from being sent.